Misc changes

fingolfin opened the pull request.

markuspf commented:
Can you explain why you made parts of the `TypOutputFile` and `TypInputFile` bitfields? For saving space in GAPState?

fingolfin commented:
Yes, that's the reason -- in fact the idea was to add a few more members in, e.g. the position in the current line doesn't need full 64 / 32 bits etc.
But given that the buffers in these structs make up the vast majority of their size anyway, this might be a pointless "optimization" (and might indeed slow down access?), so I'd be happy to drop this from this PR (or move it to a separate PR, or whatever)

fingolfin commented:
As always, if people are uncomfortable reviewing this random bunch of commits in a single PR, let me know, and I can try to split it up.

ChrisJefferson commented on src/code.c:
```diff
@@ -119,26 +119,31 @@ static inline void PopLoopNesting( void ) {
   STATE(LoopNesting) = STATE(LoopStack)[--STATE(LoopStackCount)];
 }
 
-static inline void SetupGapname(TypInputFile* i)
```
Did you benchmark this? SetupGapname gets called every time we call NewStat, and you are now doing an ELM_LIST on every call.

ChrisJefferson replied:
If you want to go down rewriting this code, you could switch back to returning void, and just move the ELM_LIST into a different tiny function, for the places where we want it (that function could call SetupGapname to ensure the values have all been installed).

fingolfin replied:
I didn't realize it was also being called by NewStat. I'll change it.

ChrisJefferson commented on src/scanner.c:
```diff
@@ -3045,7 +3045,6 @@ static Int InitKernel (
       StillMoreCookie[i][3] = 'n';  StillMoreCookie[i][4] = 'a';  StillMoreCookie[i][5] = 'm';
       StillMoreCookie[i][6] = 'e';  StillMoreCookie[i][7] = '0'+i;
       StillMoreCookie[i][8] = '\0';
-      InitGlobalBag(&(STATE(InputFiles)[i].gapname), &(StillMoreCookie[i][0]));
```
If you are getting rid of this, you don't need `StillMoreCookie` anymore.

ChrisJefferson commented on src/code.c:
```diff
@@ -109,24 +109,25 @@ static inline void PopLoopNesting( void ) {
+        Obj pos = POS_LIST(FilenameCache, filename, INTOBJ_INT(1));
+        if (pos == Fail) {
+            i->gapnameid = PushPlist(FilenameCache, filename);
+            ;
```
Why is there a ; here?

fingolfin commented:
I moved parts of this PR to two new PRs, #1833 and #1834, and also dropped the bitfield usage in `TypOutputFile` and `TypInputFile`. That should make it easier to review the remaining pieces in this PR.
